src/LegacyPaymentGateways/Adapters/LegacyPaymentGatewayAdapter.php

```diff
@@ -36,23 +36,26 @@ public function getLegacyFormFieldMarkup($formId, $registeredGateway)
      *
      * @unreleased
      *
-     * @param  array  $request  Donation Data
+     * @param  array  $legacyDonationData  Legacy Donation Data
      * @param  PaymentGatewayInterface  $registeredGateway
      *
      * @return void
      */
-    public function handleBeforeGateway($request, $registeredGateway)
+    public function handleBeforeGateway($legacyDonationData, $registeredGateway)
     {
-        $formData = FormData::fromRequest($request);
+        $formData = FormData::fromRequest($legacyDonationData);
 
         $this->validateGatewayNonce($formData->gatewayNonce);
 
         $donationId = $this->createPayment($formData->toGiveInsertPaymentData());
 
         $gatewayPaymentData = $formData->toGatewayPaymentData($donationId);
 
-        if (function_exists('Give_Recurring') && Give_Recurring()->is_recurring($formData->formId)) {
-            $subscriptionData = SubscriptionData::fromRequest($request);
+        if (
+            function_exists('Give_Recurring') &&
+            Give_Recurring()->is_donation_recurring($formData->legacyDonationData)
```

Comment:
I really don't like how this function works but I understand why we need to use it at the moment. Once we migrate the subscription concepts to core we should really simplify these recurring functions.